include/RevProc.h

```diff
@@ -156,6 +158,37 @@ class RevProc{
   ///< RevProc: Get pointer to Load / Store queue used to track memory operations
   std::shared_ptr<std::unordered_map<uint64_t, MemReq>> GetLSQueue(){ return LSQueue; }
 
+  ///< RevProc: Add a co-processor to the RevProc
+  void SetCoProc(RevCoProc* coproc);
+
+//--------------- External Interface for use with Co-Processor -------------------------
+  ///< RevProc: Allow a co-processor to manipulate the scoreboard by setting a bit. Note the RevProcPassKey may only
+  ///  be created by a RevCoProc (or a class derived from RevCoProc) so this funciton may not be called from even within
+  ///  RevProc
+  void ExternalDepSet(RevProcPasskey<RevCoProc>, uint16_t HartID, uint16_t RegNum, bool isFloat, bool value = true){
+    DependencySet(HartID, RegNum, isFloat, value);
+  }
+
+  ///< RevProc: Allow a co-processor to manipulate the scoreboard by clearing a bit. Note the RevProcPassKey may only
+  ///  be created by a RevCoProc (or a class derived from RevCoProc) so this funciton may not be called from even within
+  ///  RevProc 
+  void ExternalDepClear(RevProcPasskey<RevCoProc>, uint16_t HartID, uint16_t RegNum, bool isFloat){
+    DependencyClear(HartID, RegNum, isFloat);
+  }
```

Comment:
Note that `DependencyClear` is written in terms of `DependencySet(..., value = true)` by calling `DependencySet(..., false)`, to avoid copy-and-paste code duplication.